lib/rubygems/ext/builder.rb

```diff
@@ -48,7 +48,7 @@ def self.make(dest_path, results, make_dir = Dir.pwd)
     end
   end
 
-  def self.run(command, results, command_name = nil, dir = Dir.pwd)
+  def self.run(command, results, command_name = nil, dir = Dir.pwd, env = {})
```

Comment:
When cross-compiling with multiple ruby versions present, it's necessary to pass information about `RbConfig` to cargo so we can ensure proper linking. I found using environment variables is a simple and effective way to do this.

Reply:
If I understand it well, this argument is not used directly by `RubyGems` now. Can you confirm?

Reply:
As far as I understand, all RubyGems does is pass on the given environment to `cargo`. This feels nit to me, because it's more explicit than setting up the environment globally just so it is inherited by the `cargo` subprocess.

Reply:
RubyGems does not use this, other than Cargo builder. It allow us to pass env variables without setting them globally for the entire process.